nexus/src/app/background/sync_service_zone_nat.rs

```diff
@@ -0,0 +1,233 @@
+        Self { datastore, dpd_clients }
+    }
+}
+
+impl BackgroundTask for ServiceZoneNatTracker {
+    fn activate<'a, 'b, 'c>(
+        &'a mut self,
+        opctx: &'b OpContext,
+    ) -> BoxFuture<'c, serde_json::Value>
+    where
+        'a: 'c,
+        'b: 'c,
+    {
+        async {
+            let log = &opctx.log;
+
+            // check inventory
+            let inventory = match self
+                .datastore
+                .inventory_get_latest_collection(
+                    opctx,
+                    NonZeroU32::new(u32::MAX).unwrap(),
+                )
+                .await
+            {
+                Ok(inventory) => inventory,
+                Err(e) => {
+                    warn!(
+                        &log,
+                        "failed to collect inventory";
+                        "error" => format!("{:#}", e)
+                    );
+                    return json!({
+                        "error":
+                            format!(
+                                "failed collect inventory: \
+                                {:#}",
+                                e
+                            )
+                    });
+                }
+            };
+
+            // generate set of Service Zone NAT entries
+            let collection = match inventory {
+                Some(c) => c,
+                None => {
+                    warn!(
+                        &log,
+                        "inventory collection is None";
+                    );
+                    return json!({
+                        "error": "inventory collection is None"
+                    });
+                }
+            };
+
+            let mut ipv4_nat_values: Vec<Ipv4NatValues> = vec![];
+
+            for (sled_id, zones_found) in collection.omicron_zones {
+                let (_, sled) = match LookupPath::new(opctx, &self.datastore)
+                    .sled_id(sled_id)
+                    .fetch()
+                    .await
+                    .context("failed to look up sled")
+                {
+                    Ok(result) => result,
+                    Err(e) => {
+                        warn!(
+                            &log,
+                            "inventory collection is None";
+                            "error" => ?e,
+                        );
+                        continue;
+                    }
+                };
+
+                let sled_address = external::Ipv6Net(
+                    ipnetwork::Ipv6Network::new(*sled.ip, 128).unwrap(),
+                );
+
+                let zones_config: sled_agent_client::types::OmicronZonesConfig =
+                    zones_found.zones;
+                let zones: Vec<sled_agent_client::types::OmicronZoneConfig> =
+                    zones_config.zones;
+                for zone in zones {
+                    let zone_type: OmicronZoneType = zone.zone_type;
+                    match zone_type {
```

Comment:
I considered matching against every zone explicitly instead of using `_` to provide some sort of signal in that situation, I guess I'll implement it that way so they at least get a compiler error.